presto-main/src/test/java/io/prestosql/server/protocol/TestQueryResultRows.java

```diff
@@ -281,7 +288,63 @@ public void shouldHandleNullTimestamps()
     }
 
     @Test
-    public void shouldHandleRowTypeWithNullFieldValue()
+    public void shouldHandleNullValuesInArray()
+    {
+        List<Column> columns = ImmutableList.of(new Column("_col0", ARRAY, new ClientTypeSignature(ARRAY)));
+        List<Type> types = ImmutableList.of(new ArrayType(TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE));
+
+        List<Page> pages = rowPagesBuilder(types)
+                .row(singletonList(null))
+                .build();
+
+        TestExceptionConsumer exceptionConsumer = new TestExceptionConsumer();
+        QueryResultRows rows = queryResultRowsBuilder(getSession())
+                .withColumnsAndTypes(columns, types)
+                .withExceptionConsumer(exceptionConsumer)
+                .addPages(pages)
+                .build();
+
+        assertThat(exceptionConsumer.getExceptions()).isEmpty();
+        assertFalse(rows.isEmpty(), "rows are empty");
```

Comment:
Is this redundant with the check below

Reply:
@dain I'd like to refactor this test as some parts can be reused. This test matches others in terms of assertions so I'd leave that as it is for now if you're ok with that 🙂

Reply:
This is not an optional :)